playbooks/roles/rg_instructor_analytics_log_collector/tasks/main.yml

```diff
@@ -18,6 +18,9 @@
     name: '{{ common_web_user }}'
     groups: adm
     append: yes
+  notify:
+    - stop all processes
+    - restart supervisor
```

Comment:
in this case the order of handlers execution is unpredictable. please use the one `handler name` and this `handlers/mail.yml` task:
```
---
- name: restart supervisor
  shell: "{{ supervisor_ctl }} stop all"

- name: restart supervisor 
  service:
    name: supervisor
    state: restarted 
```

Reply:
> "Notify handlers are always run in the same order they are defined, not in the order listed in the notify-statement. This is also the case for handlers using listen."

> "If two handler tasks have the same name, only one will run. "

[Ansible Documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_intro.html)

Reply:
looks like you are right